### PR TITLE
Filter null values from being sent to graphite

### PR DESCRIPTION
**Pull request by commuterjoy**
We've a zillion metrics in hostgraphite without any values, this stops them being sent. Also added a simple test. Need to add more at some point.

/cc @matthew-andrews @wheresrhys

### CODE REVIEW

**Comment by wheresrhys** [comment]
Can we flush all the past data from graphite too in order to declutter the
graphite graph maker ui?

On 3 May 2015 at 10:36, Matt Chadburn <notifications@github.com> wrote:

> commuterjoy wants to merge 1 commit into master from no-nulls:
>
> We've a zillion metrics in hostgraphite without any values, this stops
> them being sent. Also added a simple test. Need to add more at some point.
>
> /cc @matthew-andrews <https://github.com/matthew-andrews> @wheresrhys
> <https://github.com/wheresrhys>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Financial-Times/next-metrics/pull/26
> Commit Summary
>
>    - Filter null values from being sent to graphite
>
> File Changes
>
>    - *M* Makefile
>    <https://github.com/Financial-Times/next-metrics/pull/26/files#diff-0>
>    (4)
>    - *M* lib/graphite/client.js
>    <https://github.com/Financial-Times/next-metrics/pull/26/files#diff-1>
>    (8)
>    - *M* package.json
>    <https://github.com/Financial-Times/next-metrics/pull/26/files#diff-2>
>    (5)
>    - *A* tests/graphite/clientSpec.js
>    <https://github.com/Financial-Times/next-metrics/pull/26/files#diff-3>
>    (27)
>
> Patch Links:
>
>    - https://github.com/Financial-Times/next-metrics/pull/26.patch
>    - https://github.com/Financial-Times/next-metrics/pull/26.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Financial-Times/next-metrics/pull/26>.
>

-- 

------------------------------
This email was sent by a company owned by Pearson plc, registered office at 
80 Strand, London WC2R 0RL.  Registered in England and Wales with company 
number 53723.

**Comment by commuterjoy** [comment]
@wheresrhys yeah, I think hostedgraphite deletes old keys after 14 days of inactivity. So quickest thing to do it just wait a few weeks.

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,9 @@
+
 install:
 	origami-build-tools install
+
 test:
-	next-build-tools verify;
```
Comment (matthew-andrews) [inline Makefile]: finally got bored of being told off for whitespace? :laughing:

Comment (commuterjoy) [inline Makefile]: hah, oops. It's so slow that I sometimes disable it. Let me put that back :)

Comment (commuterjoy) [inline Makefile]: @matthew-andrews ok, finally got around to jshint'ing everything. should be mergable now.

**Pull request merged**